### PR TITLE
Add nix flake

### PR DESCRIPTION
**Pull request by Dich0tomy**
This PR introduces a nix flake & default.nix that builds this rust package which allows nix users to use this package.

For people who want to use this *right now*, you can refer to this code as follows:
```nix
github:ynqa/sig/pull/1/head#default
```
Or use my fork directly before merge:
```nix
github:Dich0tomy/sig/feat/add-nix-build-method
```
But note that this ref is volatile and the branch will get deleted after this PR gets merged.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,6 +37,45 @@ brew install ynqa/tap/sigrs
...
+```
+
+Create a shell with it:
+```nix
+nix shell gituhb:ynqa/sig
+```
+
+Or run it directly:
+```nix
+cat README.md | nix run gituhb:ynqa/sig -- --archived
```
Comment (ynqa) [inline README.md:56]: Probably typo?

```suggestion
cat README.md | nix run github:ynqa/sig -- --archived
```

File: README.md [hunk 2]
```diff
@@ -37,6 +37,45 @@ brew install ynqa/tap/sigrs
...
+Add it as an input to your flake:
+```nix
+inputs = {
+  sig.url = 'github:ynqa/sig/<optional-ref>'
+}
+```
+
+Create a shell with it:
+```nix
+nix shell gituhb:ynqa/sig
```
Comment (ynqa) [inline README.md:51]: Probably typo?

```suggestion
cat README.md | nix run github:ynqa/sig -- --archived
```

**Review by ynqa — COMMENTED** [review]
@Dich0tomy Thanks for contributing, I'm not very familiar with nix yet, but I have left a comment 📝

File: README.md (hunk 1 above)
Comment (Dich0tomy) [reply README.md]: Yup, typo :sweat_smile:

File: README.md (hunk 2 above)
Comment (Dich0tomy) [reply README.md]: Same here, my bad :sweat_smile:

**Comment by Dich0tomy** [comment]
I am willing to maintain the Nix code as well.

If you add shell completions, man docs, libraries that rely on system libs (like freetype and such), somebody requests a nix shell and such, ping me and Ill make appropriate changes.

File: default.nix [hunk 3]
```diff
@@ -11,13 +10,9 @@ in
...
 
-    src = fetchFromGitHub {
-      inherit owner repo;
-      rev = "v${version}";
-      hash = "sha256-KHXBeQFmuA3YO9AN5dkY/fl/z2RdbR6AqSSEGUNrxt4=";
-    };
+    src = ./.;
 
-    cargoHash = "sha256-3nNdCz6Yjnnle/sG6kqGCAcoVRNhc4BOteWGDDl5oWs=";
+		cargoHash = "sha256-yz/DPJJxtsHpJLpTMAYfsq9miIs48F+FeSnmkQ707uA=";
```
Comment (ynqa) [inline default.nix:15]: need such indents??

File: default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  owner = "ynqa";
+  repo = "sig";
+in
+  rustPlatform.buildRustPackage {
+    pname = "sig";
+    inherit version;
+
+    src = ./.;
+
+		cargoHash = "sha256-yz/DPJJxtsHpJLpTMAYfsq9miIs48F+FeSnmkQ707uA=";
```
Comment (ynqa) [inline default.nix:15]: @Dich0tomy need such indents?

**Comment by ynqa** [comment]
> If you add shell completions, man docs, libraries that rely on system libs (like freetype and such), somebody requests a nix shell and such, ping me and Ill make appropriate changes.

@Dich0tomy  Cool. That's very helpful, thank you!

File: default.nix (hunk 4 above)
Comment (Dich0tomy) [reply default.nix:15]: Nope, but I added the blank lines for clarity. I can remove them if u want.

Comment (ynqa) [reply default.nix:15]: > I added the blank lines for clarity

I understand your intention, thank you. Let's go ahead with it as is.

**Review by ynqa — APPROVED** [review]
LGTM!

**Comment by k3d3** [comment]
Nice! I can't believe the timing on this - I was just thinking about adding this to my Nix config. Thanks adding this, Dich0tomy!

**Comment by k3d3** [comment]
Just a note, the cargo hash appears to be incorrect:

image

This is with adding the repo as an input to my flake.

**Pull request merged**